gen/AArch64Arch_gen.ml

```diff
@@ -383,7 +384,11 @@ let is_ifetch a = match a with
         let r = f (Acq o) r in
         let r = f (AcqPc o) r in
         let r = f (Rel o) r in
-        let r = f Instr r in
+        let r = 
+          if do_self then
+            let r = f Instr r in
+            r
+          else r in
```

Comment:
As the `Instr` annotation does not uses the the `o` argument it would be cleaner not to have it here:
```suggestion
```
Instead let us define:
```OCaml
let fold_self f r = if do_self then f Instr r else r
```
And call `fold_self` from `fold_acc` just below.

Reply:
thanks a lot! I will implement this into the PR